packages/ra-no-code/src/ui/ImportResourceDialog.tsx

```diff
@@ -0,0 +1,138 @@
+import { Record, useDataProvider, useRefresh } from 'ra-core';
+import { useHistory } from 'react-router-dom';
+
+export const ImportResourceDialog = (props: ImportResourceDialogProps) => {
+    const dataProvider = useDataProvider();
+    const [parsing, setParsing] = useState(false);
+    const [file, setFile] = useState<File>();
+    const [resource, setResource] = useState<string>();
+    const { addResource } = useResourceConfigurations();
+    const history = useHistory();
+    const refresh = useRefresh();
+
+    const onDrop = (acceptedFiles: File[]) => {
+        if (acceptedFiles.length > 0) {
+            const acceptedFile = acceptedFiles[0];
+            if (acceptedFile) {
+                setFile(acceptedFile);
+                setResource(acceptedFile.name.split('.')[0]);
+            }
+        }
+    };
+
+    const handleClose = () => {
+        if (props.onClose) {
+            props.onClose();
+        }
+    };
+
+    const handleSubmit = (event: FormEvent) => {
+        event.preventDefault();
+
+        if (resource && file) {
+            setParsing(true);
+            parse<Record>(file, {
+                header: true,
+                complete: ({ data }) => {
+                    data.forEach(record => {
+                        if (record.id) {
+                            dataProvider.create(resource, { data: record });
+                        }
+                    });
+                    setParsing(false);
+                    const fields = getFieldDefinitionsFromRecords(data);
+                    addResource({ name: resource, fields });
+                    history.push(`/${resource}`);
+                    handleClose();
+                    refresh();
+                },
+            });
+        }
+    };
+
+    const { getRootProps, getInputProps } = useDropzone({
+        onDrop,
+    });
+    const { ref, ...rootProps } = getRootProps();
+
+    return (
+        <Dialog
+            {...props}
+            aria-labelledby="import-resource-dialog-title"
+            aria-describedby="import-resource-dialog-description"
+        >
+            <form onSubmit={handleSubmit}>
+                <DialogTitle id="import-resource-dialog-title">
+                    Import a new resource
+                </DialogTitle>
+                {parsing ? (
+                    <DialogContent>
+                        <DialogContentText id="alert-dialog-description">
+                            Generating the resource interfaces, please wait...
+                        </DialogContentText>
+                    </DialogContent>
+                ) : (
+                    <>
+                        <RootRef rootRef={ref}>
+                            <DialogContent {...rootProps}>
+                                <input {...getInputProps()} />
+                                <DialogContentText id="alert-dialog-description">
+                                    Welcome to React Admin! Let's generate your
```

Comment:
It's not a generator, it's an app..

How about something neutral like: "Welcome to react-admin no-code"?